x/docs/internal/keeper/keeper_test.go

```diff
@@ -190,125 +190,173 @@ func TestKeeper_ShareDocument_EmptyLists(t *testing.T) {
-	assert.Contains(t, newReceivedDocs, newDocument)
+	assert.Contains(t, newReceivedDocs, newDocument.Uuid)
+}
+
+func TestKeeper_ShareDocument_SameDocumentDifferentUuid(t *testing.T) {
+	cdc, ctx, k := SetupTestInput()
+	documentsIds := types.DocumentIds{TestingDocument.Uuid}
+
+	store := ctx.KVStore(k.StoreKey)
+	store.Set(k.getSentDocumentsStoreKey(TestingSender), cdc.MustMarshalBinaryBare(&documentsIds))
+	store.Set(k.getReceivedDocumentsStoreKey(TestingRecipient), cdc.MustMarshalBinaryBare(&documentsIds))
+
+	newDocument := types.Document{
+		Uuid:       TestingDocument.Uuid + "new",
+		ContentUri: TestingDocument.ContentUri,
+		Metadata:   TestingDocument.Metadata,
+		Checksum:   TestingDocument.Checksum,
+	}
+	err := k.ShareDocument(ctx, TestingSender, ctypes.Addresses{TestingRecipient}, newDocument)
+	assert.Nil(t, err)
+
+	sentDocsBz := store.Get(k.getSentDocumentsStoreKey(TestingSender))
+	receivedDocsBz := store.Get(k.getReceivedDocumentsStoreKey(TestingRecipient))
+
+	var sentDocs, receivedDocs types.DocumentIds
+	cdc.MustUnmarshalBinaryBare(sentDocsBz, &sentDocs)
+	cdc.MustUnmarshalBinaryBare(receivedDocsBz, &receivedDocs)
+
+	assert.Equal(t, 2, len(sentDocs))
+	assert.Contains(t, sentDocs, TestingDocument.Uuid)
+
+	assert.Equal(t, 2, len(receivedDocs))
+	assert.Contains(t, receivedDocs, TestingDocument.Uuid)
 }
 
 func TestKeeper_GetUserReceivedDocuments_EmptyList(t *testing.T) {
 	_, ctx, k := SetupTestInput()
 	store := ctx.KVStore(k.StoreKey)
-	store.Delete([]byte(types.ReceivedDocumentsPrefix + TestingRecipient.String()))
+	store.Delete(k.getReceivedDocumentsStoreKey(TestingRecipient))
+
+	receivedDocs, err := k.GetUserReceivedDocuments(ctx, TestingRecipient)
+	assert.Nil(t, err)
 
-	receivedDocs := k.GetUserReceivedDocuments(ctx, TestingDocument.Sender)
 	assert.Equal(t, 0, len(receivedDocs))
 }
 
 func TestKeeper_GetUserReceivedDocuments_NonEmptyList(t *testing.T) {
 	cdc, ctx, k := SetupTestInput()
-	documents := types.Documents{TestingDocument}
 	store := ctx.KVStore(k.StoreKey)
-	store.Set([]byte(types.ReceivedDocumentsPrefix+TestingRecipient.String()), cdc.MustMarshalBinaryBare(&documents))
-	receivedDocs := k.GetUserReceivedDocuments(ctx, TestingRecipient)
+
+	documents := types.Documents{TestingDocument}
+	documentIds := types.DocumentIds{TestingDocument.Uuid}
+
+	store.Set(k.getDocumentStoreKey(TestingDocument.Uuid), cdc.MustMarshalBinaryBare(TestingDocument))
+	store.Set(k.getReceivedDocumentsStoreKey(TestingRecipient), cdc.MustMarshalBinaryBare(&documentIds))
+
+	receivedDocs, err := k.GetUserReceivedDocuments(ctx, TestingRecipient)
+	assert.Nil(t, err)
 
 	assert.Equal(t, 1, len(receivedDocs))
 	assert.Equal(t, documents, receivedDocs)
 }
 
 func TestKeeper_GetUserSentDocuments_EmptyList(t *testing.T) {
 	_, ctx, k := SetupTestInput()
+
 	store := ctx.KVStore(k.StoreKey)
-	store.Delete([]byte(types.SentDocumentsPrefix + TestingSender.String()))
+	store.Delete(k.getSentDocumentsStoreKey(TestingSender))
```

Comment:
you shouldn't need this if you keep initializing store in every test with new `ctx` and ´k`